Change values of standard colours to follow CSS standard

vadz opened the pull request:
This implements my proposal from #23518. It is obviously backwards-incompatible but I've tried to make it as compatible as possible and hope this should be good enough. Please try to test your applications with this PR and let me know if you see something really bad and/or if using `UseScheme(Traditional)` doesn't work for you to restore the original appearance.

@paulcor Do you think we should remove wxGTK-specific `wxColour::FromString()` now? The effect of doing it would be to make wxGTK use the same colours as wxMSW used to use if `UseScheme(Traditional)` is called, while continuing to use the same colours as today by default. I'm not 100% if this is desirable, but I think it might be.

vadz committed:
Delete wxTheColourDatabase in wxDeleteStockLists() too

No real changes, just remove special handling of wxTheColourDatabase
which was allocated, together with the other global lists, in
wxInitializeStockLists() but deleted in wxAppBase::CleanUp() from where
wxDeleteStockLists() was called instead of in that function itself.

Remove this discrepancy.

vadz committed:
Use ASCII string for standard colour database colour names

Don't bother with wxT() when we can be sure that these names only
contain ASCII.

No real changes.

vadz committed:
Simplify handling of GREY/GRAY in wxColourDatabase

Just add the GRAY variants to the database directly instead of checking
for both GREY and GRAY when looking the colours up.

Note that we don't automatically add the version using GREY if a colour
with GRAY in it is added explicitly via AddColour() as this is not
something that had ever been documented and doesn't seem to make much
sense: application programmer using "GRAY" when adding a colour is
unlikely to use "GREY" when retrieving it later.

vadz committed:
Don't use wxColourDatabase for stock pen/brush initialization

This makes things simpler, removing an interdependence between
wxStockGDI and wxColourDatabase, and isolates the grey and medium grey
pens and brushes from the upcoming change to the value of the latter
colour in wxColourDatabase.

Note that we intentionally don't add wxGREY and wxMEDIUM_GREY macros,
unlike for the existing stock colours, as they are not really necessary
and it doesn't seem worth to add them just for consistency, especially
because it's easy to imagine some existing code already defining them,
especially the former one.

No real changes yet.

vadz committed:
Show standard colours in alphabetical order in "drawing" sample

This is nicer than showing them in the apparently random order in which
they appear in the hash table.

vadz committed:
Use CSS colour values in wxColourDatabase by default

Allow reverting to using the traditional values for the conflicting
colours by calling UseScheme(Traditional) if really needed.

This is another attempt for changing wxColourDatabase to use more
standard colour values, after the previous attempt in bb131fdbc5
(updated colour definitions to the new official X version (patch
771272), 2003-07-17) which was reverted almost immediately after in
5e2ab1ea5d (reverted the last change (colour values changes), it cerated
too many problems, 2003-07-19), see #6031. Unlike the previous ones, it
provides an escape hatch in the form of UseScheme() and also leaves
traditional wx colour names not clashing with CSS ones still available.

Also update the "drawing" sample to allow showing both colour schemes,
and extend its view port to fit all the colour names vertically.

See #23518.

vadz committed:
Add simple wxColourDatabase unit test

Notably check that the recently added UseScheme() works.

vadz committed:
Optimize wxColourDatabase initialization by not creating colours

We don't need to create ~200 colours of which we're probably only going
to use only a few on program startup, so don't do it and really create
them only when they're really needed.

paulcor commented:
> Do you think we should remove wxGTK-specific wxColour::FromString() now?

Yes, I think it's better to just do it one way.

vadz committed:
Stop using GDK functions for parsing colours names

This shouldn't be necessary now as they use the same CSS colour names
that we use too by default and like this switching to "Traditional"
colour names also changes wxGTK behaviour to use wxMSW colour values.
